ovh/resource_cloud_project_network_private_subnet_v2.go

```diff
@@ -0,0 +1,267 @@
+package ovh
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ovh/terraform-provider-ovh/ovh/helpers"
+)
+
+func resourceOvhCloudProjectNetworkPrivateSubnetV2ImportState(
+	d *schema.ResourceData,
+	meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	splitId := strings.SplitN(givenId, "/", 3)
+	if len(splitId) != 3 {
+		return nil, fmt.Errorf("Import Id is not service_name/network_id/subnet_id formatted")
+	}
+	d.SetId(splitId[2])
+	d.Set("network_id", splitId[1])
+	d.Set("service_name", splitId[0])
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	log.Printf(
+		"[DEBUG] Will Import ovh_cloud_project_network_private_subnet with project %s, network %s, id %s",
+		d.Get("service_name"),
+		d.Get("network_id"),
+		d.Id(),
+	)
+	return results, nil
+}
+
+func resourceCloudProjectNetworkPrivateSubnetV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudProjectNetworkPrivateSubnetV2Create,
+		Read:   resourceCloudProjectNetworkPrivateSubnetV2Read,
+		Delete: resourceCloudProjectNetworkPrivateSubnetV2Delete,
+		Importer: &schema.ResourceImporter{
+			State: resourceOvhCloudProjectNetworkPrivateSubnetV2ImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OVH_CLOUD_PROJECT_SERVICE", nil),
+				Description: "Service name of the resource representing the id of the cloud project.",
+			},
+			"network_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"dhcp": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+			"cidr": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: resourceCloudProjectNetworkPrivateSubnetValidateNetwork,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"gateway_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"enable_gateway_ip": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+			"dns_nameservers": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: resourceCloudProjectNetworkPrivateSubnetValidateIP,
+				},
+				Description: "List of DNS nameservers, default: 213.186.33.99",
+			},
+			"host_routes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
```

Comment:
why don't you define the nested object ?

Reply:
how ? it is not a standalone object in API / terraform, I already try with a full resource definition it does not work...